### PR TITLE
Sanitise step descriptions

### PR DESCRIPTION
**Pull request by i386**
# Description

@abayer @vivek looks like `ArgumentsAction.getStepArgumentsAsString` might include `\t` and other characters that are bad for JSON. Should this really be updated in workflow-api? I am not sure having a `\t` character in a step name is something good for human eyeballs anyway. Feels some sanitation should occur there rather in blue ocean?

The problem is that when we encounter these issues parts of the UI fail to load as described in the ticket below.

See [JENKINS-45099](https://issues.jenkins-ci.org/browse/JENKINS-45099).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
nice - would like the nod from @vivek on this that this is right place and if there are other problems.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java [hunk 1]
```diff
@@ -74,7 +75,12 @@ public String getDisplayName() {
 
     @Override
     public String getDisplayDescription() {
-        return ArgumentsAction.getStepArgumentsAsString(node.getNode());
+        String displayDescription = ArgumentsAction.getStepArgumentsAsString(node.getNode());
+        if (displayDescription != null) {
+            // JENKINS-45099 Remove any control characters that may have found their way out of a script
+            displayDescription = CharMatcher.JAVA_ISO_CONTROL.and(CharMatcher.anyOf("\r\n\t").negate()).removeFrom(displayDescription);
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: I wonder if there is a more general sanitise thing that can be done here...

Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: I am not sure myself. Advice needed.

Comment (vivek) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: @i386 Code looks good, if I read it correctly, it filters out ISO controls chars (not json friendly) and line/tab breaks. We should keep it inside commons package, maybe inside Utils.java as generic code, could be used somewhere else if needed, perhaps something along this?

```
public String sanitizeJsonString(@Nonnull String value, @Nullable String additionalFilter){
        
        CharMatcher charMatcher = CharMatcher.JAVA_ISO_CONTROL;
        if(StringUtils.isNotBlank(additionalFilter)){
            charMatcher = charMatcher.and(CharMatcher.anyOf(additionalFilter).negate());
        }
        return charMatcher.removeFrom(value);
    }
```

Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepImpl.java]: Done.

File: blueocean-commons/src/test/java/io/jenkins/blueocean/commons/JSONTest.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.jenkins.blueocean.commons;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class JSONTest {
```
Comment (i386) [inline blueocean-commons/src/test/java/io/jenkins/blueocean/commons/JSONTest.java]: @vivek added a test for the util

**Pull request merged**